[reland][export] Fix graph signature for primitive outputs

angelayi opened the pull request:
Summary: Reland of D53233649/https://github.com/pytorch/pytorch/pull/118655. Previously I didn't realize there was a use-case of a torchbind object as an input to the graph, so I didn't mark `CustomObjArgument` as a valid input, which broke [this test](https://github.com/pytorch/pytorch/blob/a43c28368c184ba1bf964f4fb99bec300917e2f4/test/export/test_torchbind.py#L81). Somehow the initial CI did not catch it, but hud was sad so that PR was reverted. So now I added `CustomObjArgument` as valid input [here](https://github.com/pytorch/pytorch/pull/118818/files#diff-92420f977c3a02b2deadf6752ce4a9ee601c20612a1a13cc365252eb09410edbR298).

Test Plan: CI

Reviewed By: tarun292

Differential Revision: D53288445

ydwu4 commented on test/export/test_export.py:
```diff
@@ -2985,6 +2985,15 @@ def forward(self, l_q_, l_k_, l_v_):
     getitem = _scaled_dot_product_flash_attention[0];  _scaled_dot_product_flash_attention = None
     return (getitem,)""")
 
+    def test_int_list_output(self):
```
Why the test is for list_output instead of user inputs? I thought the code change is for inputs

angelayi replied:
original fix was for the handling outputs. and then i decided to fix the inputs the same way so that we know clearly the types of inputs allowed...but that caused a test failure.

angelayi replied:
ohh i understand... i fixed the merge conflict badly

ydwu4 replied:
ok, probably add a test for input as well for symetry?

angelayi committed:
[reland][export] Fix graph signature for primitive outputs (#118818)

Summary:

Reland of D53233649

Test Plan: CI

Reviewed By: tarun292

Differential Revision: D53288445
